### PR TITLE
set up goreleaser with github action

### PR DESCRIPTION
**Pull request by yihanzhen**
Tested in [my fork](https://github.com/yihanzhen/go-containerregistry/releases).

To make it work we'll also need to add a github secret `RELEASE_TOKEN` (which requires admin permission to the repo).

### CODE REVIEW

**Comment by yihanzhen** [comment]
@jonjohnsonjr @ImJasonH PTAL

**Comment by yihanzhen** [comment]
cc/ @lindsayismith

**Comment by lindsayismith** [comment]
What is the name of the release binary that it creates?  How does it relate to the tags we push?

**Comment by yihanzhen** [comment]
@lindsayismith The name of the binary is `go-containerregistry_{version}_{os}.tar.gz`. The tag gets filled into the `{version}` part. See [this](https://github.com/yihanzhen/go-containerregistry/releases/tag/v0.4.0) as an example.

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This is an example goreleaser.yaml file with some sane defaults.
+# Make sure to check the documentation at http://goreleaser.com
+# before:
+#   hooks:
+#     # You may remove this if you don't use go modules.
+#     - go mod download
+#     # you may remove this if you don't need go generate
+#     - go generate ./...
+builds:
+- id: registry
```
Comment (jonjohnsonjr) [inline .goreleaser.yml:10]: Let's drop the registry binary from this -- it's primarily for testing and I don't want to implicitly endorse its use by including it in releases.

Comment (yihanzhen) [reply .goreleaser.yml]: Done

**Pull request merged**